verus/lcf/src/verusport.rs

```diff
@@ -0,0 +1,1119 @@
+use crate::string_hash_map::TmpStringHashMap;
+use vstd::assert_seqs_equal;
+use vstd::prelude::*;
+use vstd::seq::Seq;
+
+verus! {
+
+pub enum Const {
+    Atom(String),
+    Nat(u64),
+    Str(String),
+}
+
+impl PartialEq for Const {
+    //Function allows for Const types to be evaluated for equality and aid Verus verifier; not fully sufficient.
+    //Must call const_eq for Atoms and Strs
+    fn eq(&self, other: &Self) -> (res: bool)
+        ensures
+            res <==> self.deep_view() == other.deep_view(),
+    {
+        match (self, other) {
+            (Const::Atom(s), Const::Atom(t)) => s == t,
+            (Const::Nat(u), Const::Nat(v)) => u == v,
+            (Const::Str(s), Const::Str(t)) => s == t,
+            (Const::Atom(_), Const::Nat(_)) | (Const::Nat(_), Const::Atom(_)) => false,
+            (Const::Atom(_), Const::Str(_)) | (Const::Str(_), Const::Atom(_)) => false,
+            (Const::Str(_), Const::Nat(_)) | (Const::Nat(_), Const::Str(_)) => false,
+        }
+    }
+}
+
+impl Const {
+    // function uses eq from impl Partial Eq to prove equality between items
+    pub fn const_eq(&self, other: &Self) -> (res: bool)
```

Comment:
Can you explain the difference between `const_eq` and `eq`? Can one be implemented in terms of the other?

Reply:
@eobardthawne21p it might be worth adding this to the TODOs/future work section of the readme